salt/modules/mssql.py

```diff
@@ -67,7 +71,71 @@ def _get_connection(**kwargs):
             connection_args[arg] = __salt__["config.option"](
                 "mssql." + arg, _DEFAULTS.get(arg, None)
             )
-    return pymssql.connect(**connection_args)
+    conn = pymssql.connect(**connection_args)
+    conn.autocommit(True)
+    return conn
+
+
+def _close_connection(cursor=None, connect=None):
+    """
+    Close the database connection and cursor connection
+    Args:
+        cursor (cursor object):
+        connect  (Connection object):
+
+    """
+    try:
+        if cursor:
+            cursor.close()
+    except pymssql.OperationalError:
+        pass
+    try:
+        if connect:
+            connect.close()
+    except pymssql.OperationalError:
+        pass
+
+
+def _to_rawstrings(text):
+    """
+    Escape an argument string to be suitable to the appropriate caller.
+    Special characters will be escaped.
+    For example, if a Windows path is passed in, special characters need to be escaped
+
+    Args:
+        text (str): a string argument to escape
+
+    Returns:
+        str: an escaped string
+    """
+    s = ""
+    if six.PY2:
+        if isinstance(text, six.string_types):
+            s = text.encode(encoding="string-escape")
+        elif isinstance(text, six.text_type):
+            s = text.encode(encoding="unicode-escape")
+    elif six.PY3:
+        if isinstance(text, six.text_type):
+            s = text.encode(encoding="unicode-escape")
+    return s
```

Comment:
This could be simpler:
```python
s = ""
if isinstance(text, six.text_type):
    s = text.encode(encoding="unicode-escape")
elif isinstance(text, six.string_types):
    s = text.encode(encoding="string-escape")
return s
```
Your don't need to check Py version here because on Py3 text_type is str and string_types contains str only. So you'll not enter the `elif` case on Py3.

Reply:
Ok, I see